### PR TITLE
feat(pipelines): Accept Stdin for pipeline save.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

File: command/pipelines/pipeline_save_test.go [hunk 1]
```diff
@@ -46,6 +46,34 @@ func TestPipelineSave_basic(t *testing.T) {
...
+	defer os.Remove(tempFile.Name())
+
+	// Prepare Stdin for test reading.
+	tempFile.Seek(0, 0)
+	oldStdin := os.Stdin
+	defer func() { os.Stdin = oldStdin }()
+	os.Stdin = tempFile
+
+	args := []string{"--gate-endpoint", ts.URL}
+	fmt.Println(args)
```
Comment (sbwsg) [inline command/pipelines/pipeline_save_test.go]: `*testing.T` has `Log` and `Logf` functions that silence output unless -v flag is passed or the test fails.  Do you intend for this to always be printed as part of test output?

File: command/pipelines/pipeline_save.go [hunk 2]
```diff
@@ -49,11 +51,23 @@ func (c *PipelineSaveCommand) flagSet() *flag.FlagSet {
 
 // parsePipelineFile reads and deserializes the input pipeline file.
```
Comment (sbwsg) [inline command/pipelines/pipeline_save.go]: is this comment out of date now that the func falls back to stdin?  i guess stdin is technically a file.

Comment (jtk54) [reply command/pipelines/pipeline_save.go]: Yep.

File: command/pipelines/pipeline_save_test.go (hunk 1 above)
Comment (jtk54) [reply command/pipelines/pipeline_save_test.go]: That shouldn't be there in the first place.

File: command/pipelines/pipeline_save.go (hunk 2 above)
Comment (jtk54) [reply command/pipelines/pipeline_save.go]: Yep as in it's out of date and I updated the comment*

**Comment by jtk54** [comment]
Closes https://github.com/spinnaker/spinnaker/issues/3058.

**Pull request merged**